src/analysis/classic/druid/feral/modules/spells/FaerieFire.tsx

```diff
@@ -0,0 +1,71 @@
+import SPELLS from 'common/SPELLS/classic/druid';
+import { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import Abilities from 'parser/core/modules/Abilities';
+import SPELL_CATEGORY from 'parser/core/SPELL_CATEGORY';
+import ExecuteHelper from 'parser/shared/modules/helpers/ExecuteHelper';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+class FaerieFire extends ExecuteHelper {
+  static executeSources = SELECTED_PLAYER;
+  static modifiesDamage = false;
+
+  static dependencies = {
+    ...ExecuteHelper.dependencies,
+    abilities: Abilities,
+  };
+
+  maxCasts: number = 1;
+
+  activeFFFSpell = SPELLS.FAERIE_FIRE_FERAL;
+
+  protected abilities!: Abilities;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = true;
+
+    this.addEventListener(Events.fightend, this.adjustMaxCasts);
+    const ctor = this.constructor as typeof ExecuteHelper;
+    ctor.executeSpells.push(this.activeFFFSpell);
+
+    (options.abilities as Abilities).add({
+      spell: SPELLS.FAERIE_FIRE_FERAL.id,
+      category: SPELL_CATEGORY.ROTATIONAL,
+      charges: 1,
+      cooldown: 6,
+      gcd: {
+        base: 1000,
+      },
+      castEfficiency: {
+        suggestion: true,
+        recommendedEfficiency: 0.85,
+        maxCasts: () => this.maxCasts,
+      },
+    });
+  }
+
+  adjustMaxCasts() {
+    this.maxCasts += Math.ceil(this.totalExecuteDuration / 10000);
```

Comment:
this should divide by 6000 (the cooldown of FF in milliseconds) instead of 10000 (looks like its taken from Kill Shot, which has a 10s cooldown)

(also faerie fire is only usable in execute? interesting)

Reply:
Ahh I'll need to rework this, I'll probably end up making a new PR for faerie fire stuff